torchci/components/TorchAgentPage.tsx

```diff
@@ -326,8 +360,12 @@ export const TorchAgentPage = () => {
   useEffect(() => {
     if (session.data?.user) {
       fetchChatHistory();
+      // Only check permissions if we haven't checked yet
+      if (!hasCheckedPermissions && !isCheckingPermissions) {
+        checkUserPermissions();
```

Comment:
instead of boolean flags can just memoize the result with `useCallback` hook.

same with `fetchChatHistory`